### PR TITLE
Replace checked in storage in snake_oil with made on request

### PR DESCRIPTION
**Pull request by oyvindeide**
The checked in storage for snake oil was not up to date, for example it did not reflect the changes that were made to the RNG. Instead of continuously keeping it up to date, I suggest we instead create a storage on request, and rather try to move away from the pattern of using snake oil for everything.

## Pre review checklist

- [x] Added appropriate release note label
- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).

Adding labels helps the maintainers when writing release notes. This is the [list of release note labels](https://github.com/equinor/ert/labels?q=release-notes).

### CODE REVIEW

File: tests/unit_tests/test_libres_facade.py [hunk 1]
```diff
@@ -281,40 +285,42 @@ def test_summary_collector(monkeypatch, facade):
...
 
-    assert pytest.approx(data["MISFIT:TOTAL"][0]) == 767.008457
-    assert pytest.approx(data["MISFIT:TOTAL"][24]) == 1359.172803
+    assert pytest.approx(data["MISFIT:TOTAL"][0]) == 2024.461010928874
+    assert pytest.approx(data["MISFIT:TOTAL"][4]) == 831.520534770889
 
     # pylint: disable=pointless-statement
     # realization 20:
-    data.loc[20]
+    data.loc[2]
```
Comment (ManInFez) [inline tests/unit_tests/test_libres_facade.py:300]: what does this do?

File: tests/unit_tests/c_wrappers/res/enkf/data/test_gen_data.py [hunk 2]
```diff
@@ -2,10 +2,10 @@
 from ert._c_wrappers.enkf.node_id import NodeId
 
 
-def test_load_active_masks(snake_oil_case):
+def test_load_active_masks(snake_oil_case_storage):
     case1 = "default_0"
     case2 = "default_1"
```
Comment (ManInFez) [inline tests/unit_tests/c_wrappers/res/enkf/data/test_gen_data.py:7]: does not exist anymore

File: tests/unit_tests/test_libres_facade.py (hunk 1 above)
Comment (oyvindeide) [reply tests/unit_tests/test_libres_facade.py]: :shrug: `# pylint: disable=pointless-statement` ? Who knows what the intent was, I have removed it.

**Review by ManInFez — APPROVED** [review]
:bacon: n1

**Pull request merged**